Documentation: Add a page that explains how to use cmake (#4624)
CGAL/cgal: Documentation/doc/Documentation/Developer_manual/create_and_use_a_cmakelist.txt

/*!

> \page devman_create_and_use_a_cmakelist How to use %CGAL with CMake

Reviewer: ```suggestion
\page devman_create_and_use_a_cmakelist How to use \cgal with CMake
```

> This page will explain how to manually create a CMakeLists.txt file to link a custom program with CGAL.

Reviewer: ```suggestion
This page will explain how to manually create a CMakeLists.txt file to link a custom program with \cgal.
```

A base can be created using the script `cgal_create_CMakeLists`. Its usage is detailed in Section \ref devman_create_cgal_CMakeLists.

> \section seclink Linking with CGAL

Reviewer: ```suggestion
\section seclink Linking with \cgal
```

> To link with the CGAL library, use the following :

Reviewer: ```suggestion
To link with the \cgal library, use the following :
```

\code
find_package(CGAL)
add_executable(my_executable my_source_file.cpp)
target_link_libraries(my_executable CGAL::CGAL)
\endcode

Other CGAL libraries are linked similarly. For example, with CGAL_Core:

\code
find_package(CGAL REQUIRED COMPONENTS Core)
target_link_libraries(my_executable CGAL::CGAL CGAL::CGAL_Core)
\endcode

> There are also some cmake macros to link with CGAL dependenciesthat can be found in

Reviewer: ```suggestion
There are also some cmake macros to link with CGAL dependencies that can be found in
```

the section \subpage thirdparty.

\note The CGAL targets define the following compiler flags :
  - `-frounding-math` with gcc
  - `/fp:strict /fp:except-` with MSVC

\section secexample Minimal Example Using Qt5

> This section describes a minimal example of a program that uses %CGAL and Qt5 for some GUI features.

Reviewer: ```suggestion
This section describes a minimal example of a program that uses \cgal and Qt5 for some GUI features.
```

> This section describes a minimal example of a program that uses %CGAL and Qt5 for some GUI features.

Reviewer: ```suggestion
This section describes a minimal example of a program that uses \cgal and Qt5 for some GUI features.
```

/*!

> \page devman_create_and_use_a_cmakelist How to use CGAL with CMake

Reviewer 2: ```suggestion
\page devman_create_and_use_a_cmakelist How to use %CGAL with CMake
```
... to remove the automatic linking of `CGAL`.

Reviewer 2: @sloriot The initial issue (a link on "CGAL") is still there, now that you can reverted to using `CGAL`. I think it should be `%CGAL`.

See here: https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.1-Ic-127/output2/Manual/devman_create_and_use_a_cmakelist.html

Reviewer: I saw yesterday that it was breaking the navtree but actually it was another line. Fixed in 2dc7f66c8a5.

This page will explain how to manually create a CMakeLists.txt file to link a custom program with CGAL.
A base can be created using the script `cgal_create_CMakeLists`. Its usage is detailed in Section \ref devman_create_cgal_CMakeLists.
\section seclink Linking with CGAL
To link with the CGAL library, use the following :
\code
find_package(CGAL)
add_executable(my_executable my_source_file.cpp)
target_link_libraries(my_executable CGAL::CGAL)
\endcode

Other CGAL libraries are linked similarly. For example, with CGAL_Core:

\code
find_package(CGAL REQUIRED COMPONENTS Core)
target_link_libraries(my_executable CGAL::CGAL CGAL::CGAL_Core)
\endcode

There are also some cmake macros to link with CGAL dependencies. For example, to link with eigen, you can use

\code
CGAL_target_use_Eigen(<target>)
\endcode
For more details on these macros, see \subpage thirdparty.

\note The CGAL targets define the following compiler flags :
  - `-frounding-math` with gcc
  - `/fp:strict /fp:except-` with MSVC

\section secexample Minimal Example Using Qt5
This section describes a minimal example of a program that uses CGAL and Qt5 for some GUI features.

\subsection subcmake CMakeLists.txt
\code

> cmake_minimum_required(VERSION 3.1)

Reviewer 2: ```suggestion
cmake_minimum_required(VERSION 3.1...3.17)
```
